plugins/kernel_v2/config/ssh_connect.rb

```diff
@@ -134,6 +142,26 @@ def validate(machine)
             given: @connect_timeout.to_s)
         end
 
+        if !@retries.is_a?(Integer)
```

Comment:
Take a look at the [`validate` method for the winrm communicator](https://github.com/hashicorp/vagrant/blob/main/plugins/communicators/winrm/config.rb#L51). It has some better validations steps. For example, it's ok to not have any retries or to have a retry interval between 0 and 1.